### PR TITLE
Slightly nerfs and fixes slaughter demon healing

### PR DESCRIPTION
**Pull request by SchrodingersWolf**
## What Does This PR Do
This PR sets out to do two things:
1. Fixes an oversight with #18554 - Slaughter demons now heal the correct amount (25 HP) from consuming simplemobs.
2. Slaughter demons also heal less from consuming mindless carbon mobs, only gaining 50 HP instead of a full heal.

## Why It's Good For The Game
Going to address each point one by one:
1. Bugfixes are good. We don't want the slaughter demon healing almost half of their full health because they ate Ian.
2. Slaughter demons can also get a full heal off monkeys, which is an issue as they are a riskless source of healing - Monkeys can't fight back or even run, so they should heal less compared to crew.

## Images of changes
Will post soon

## Testing
Loaded in testserver. Became slaughter demon. Killed and ate Ian, a monkey, and a skrell that my alt had logged into. Did maths and verified that they healed the correct amount.

## Changelog
:cl:
tweak: Slaughter demon now heal less from mindless mobs
fix: Slaughter demon now heal the correct amount from simplemobs
/:cl:

### CODE REVIEW

**Comment by GDNgit** [comment]
Please leave the fixes and the tweaks in their own prs

File: code/datums/spells/bloodcrawl.dm [hunk 1]
```diff
@@ -131,20 +133,27 @@
...
 		to_chat(L, "<span class='warning'>You devour [victim]. Your health is fully restored.</span>")
 		L.adjustBruteLoss(-1000)
 		L.adjustFireLoss(-1000)
 		L.adjustOxyLoss(-1000)
 		L.adjustToxLoss(-1000)
+		return
+	if(mindless)
+		to_chat(L, "<span class='warning'>You devour [victim], but their lack of intelligence renders their flesh dull and unappetising, leaving you wanting for more.</span>")
+		L.adjustBruteLoss(-50)
+		if(isslaughterdemon(L))
```
Comment (GDNgit) [inline code/datums/spells/bloodcrawl.dm:150]: is this logic supposed to be reversed?

Comment (SchrodingersWolf) [reply code/datums/spells/bloodcrawl.dm:150]: Yes - I missed this during testing. My mistake. Will change it right now.

**Comment by SchrodingersWolf** [comment]
> Please leave the fixes and the tweaks in their own prs

I was misinformed about the Contributor Guidelines. I'll remove the bloodcrawl balance changes in this PR.

File: code/datums/spells/bloodcrawl.dm [hunk 2]
```diff
@@ -131,20 +132,27 @@
 		playsound(get_turf(L), sound, 100, 1)
 		sleep(3 SECONDS)
 
+	var/mindless = (ishuman(victim) || isrobot(victim))
```
Comment (lewcc) [inline code/datums/spells/bloodcrawl.dm:135]: If you're only making this check once, I'd say you don't need this variable and can instead just check the condition where you currently check the variable

Comment (SchrodingersWolf) [reply code/datums/spells/bloodcrawl.dm:135]: Yeah, that is a relic of when the code is less optimised. I'll change that in a bit.

**Review by Henri215 — APPROVED** [review]

**Pull request merged**